bundle/src/main/java/com/adobe/acs/commons/oak/impl/EnsureOakIndexJobHandler.java

```diff
@@ -482,6 +498,8 @@ boolean needsUpdate(@Nonnull Resource ensureDefinition, @Nonnull Resource oakInd
         final CustomChecksumGeneratorOptions ensureDefinitionOptions = new CustomChecksumGeneratorOptions();
         ensureDefinitionOptions.addIncludedNodeTypes(new String[]{NT_OAK_UNSTRUCTURED});
         ensureDefinitionOptions.addExcludedProperties(this.ignoreProperties);
+        ensureDefinitionOptions.addExcludedSubTrees(this.excludeSubTrees);
```

Comment:
Shouldn't these two lines also be added to `oakIndexOptions`? Also notice `this.excludeSubTrees` is based on `NT_OAK_QUERY_INDEX_DEFINITION`, which is meant for `oakIndexOptions` if I am not mistaken. Essentially we may have to have two `excludeSubTrees` based on `NT_OAK_UNSTRUCTURED` and `NT_OAK_QUERY_INDEX_DEFINITION` and added to the two `options` respectively?

Reply:
right you are! and thats why @kelvinxzf should always review my PR's! :)

Reply:
Maybe you need the following as well?
`oakIndexOptions.addExcludedSubTrees(this.excludeSubTrees);
 oakIndexOptions.addExcludedNodeNames(this.excludeNodeNames);`